### ISSUE TITLE
__init__.py: "no module _essentia"

### ISSUE THREAD

**Issue by marikgoldstein** [MTG/essentia#436]
Hi,

I have just installed essentia. I am very excited to use it, but I am having an import problem.
It is not an environmental issue, as python is able to find the module and attempts to load it.

The issue happens when the __init _.py file attempts to import "_essentia".
The error is that no module named _essentia is found (see below).

In the same folder as __init__.py, I see an "_essentia.so". I am not sure exactly how related it is to the missing "_essentia".

Any advice?

Thanks!

Mark
-------
>>> import essentia
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python2.7/site-packages/essentia/__init__.py", line 1, in <module>
    import _essentia
ImportError: No module named '_essentia'

**Comment by alastair** [comment]
Which version of swig do you have installed? This looks like the same error as https://github.com/MTG/gaia/issues/35#issuecomment-228065236

If you're using a recent version of swig (3.0.x), then you should be able to fix this by moving `/usr/local/lib/python2.7/site-packages/essentia/_essentia.so` up one level directly into `site-packages`

**Comment by marikgoldstein** [comment]
Thanks so much for the reply.

*1. *

I tried moving the essentia.so up one directory into site-packages, but it
did not help. I have the latest swig installed.

This is a stupid question... the fact that you are mentioning
python2.7/site-packages, does that mean that essentia only works for
python2.7?

*2. I can't get it to work but I have narrowed down the problem for sure:*

When I re-install essentia with homebrew, homebrew warns me that the
package is not in its search path and that I won't be able to find it. It
recomends that I do this:

* echo 'import site;
site.addsitedir("/usr/local/lib/python2.7/site-packages")' >>
/Users/pythongoldstein/.local/lib/python3.5/site-packages/homebrew.pth*

When that line *IS PLACED INTO* homebrew.pth, other libraries of mine break
and I have some sitecustomize and pythonpath problems. When I import
essentia, I have the problem I described above: it is indeed found, but
_essentia is not found:

File "/usr/local/lib/python2.7/site-packages/essentia/__init__.py", line 1,
in <module>

    import _essentia

ImportError: No module named '_essentia'

When that line *IS NOT PLACED INTO *homebrew.pth, my other libraries are
fixed but python cannot even find essentia. Meaning I don't get the problem
of no _essentia, but rather it's not able to find essentia at all.


I don't expect you to debug but my environment, but would be happy to hear
if you have any ideas.

Basically I use a conda environment. Maybe it makes sense to switch to
something else.

I would love to have both librosa and essentia working :(

THANKS
Mark

**Comment by marikgoldstein** [comment]
I have librosa in python2.7/site-packages but it works for python 3.5

**Comment by alastair** [comment]
We don't support essentia yet in python 3. It's an outstanding task which we need to complete.
I mentioned `/usr/local/lib/python2.7/site-packages` because it was in your error message. I strongly suggest that you do not mix packages from python 2's site packages into python 3's path.

**Comment by alastair** [comment]
Leaving open because this requires changes to essentia bindings to work with new version of swig

**Comment by brennon** [comment]
But @marikgoldstein, what was the fix?!? I'm finding essentia to be _impossibly_ difficult and unintuitive to install. I'd love to use it, and have worked around countless obstacles before hitting this one, and now I'm completely stuck. Has anyone found a fix or workaround for this?

**Comment by brennon** [comment]
Well...reinstalling got me past _this_ issue, but now I'm on to issue #319...

**Comment by marikgoldstein** [comment]
Hi @brennon Brennon, sorry I missed your post.

Actually, @alastair ,  I didn't fully understand your comment about reopening the thread.

Actually, my only (embarrassing) problem, was that I didn't realize Essentia was python2 only!
I had been trying to install it in Python3.

After I tried in python2, I didn't need to ANYTHING except for the basic brew installation steps!

**Comment by dbogdanov** [comment]
True, Essentia python bindings do not rely on swig. Therefore closing this issue .

**Comment by Shahriar567** [comment]
can we un essentia in python3 now ?
